Core/SaveState.cpp

```diff
@@ -471,6 +460,63 @@ namespace SaveState
 		}
 	}
 
+	void LoadSlot(const Path &gameFilename, int slot, Callback callback, void *cbUserData)
+	{
+		Path fn = GenerateSaveSlotFilename(gameFilename, slot, STATE_EXTENSION);
+		if (!fn.empty()) {
+			// This add only 1 extra state, should we just always enable it?
+			if (g_Config.bEnableStateUndo) {
+				Path backup = GetSysDirectory(DIRECTORY_SAVESTATE) / LOAD_UNDO_NAME;
+				
+				auto saveCallback = [=](Status status, const std::string &message, void *data) {
+					if (status != Status::FAILURE) {
+						DeleteIfExists(backup);
+						File::Rename(backup.WithExtraExtension(".tmp"), backup);
+						g_Config.sStateLoadUndoGame = GenerateFullDiskId(gameFilename);
+						Load(fn, slot, callback, cbUserData);
+					} else if (callback) {
+						callback(status, message, data);
+					}
+				};
+
+				if (!backup.empty()) {
+					Save(backup.WithExtraExtension(".tmp"), -2, saveCallback, cbUserData);
+				} else {
+					auto sy = GetI18NCategory("System");
+					if (callback)
+						callback(Status::FAILURE, sy->T("Failed to save state for load undo. Error in the file system."), cbUserData);
+				}
+			} else {
+				Load(fn, slot, callback, cbUserData);
+			}
+		} else {
+			auto sy = GetI18NCategory("System");
+			if (callback)
+				callback(Status::FAILURE, sy->T("Failed to load state. Error in the file system."), cbUserData);
+		}
+	}
+
+	bool UndoLoad(const Path &gameFilename, Callback callback, void *cbUserData)
+	{
+		if (g_Config.sStateLoadUndoGame != GenerateFullDiskId(gameFilename)) {
+			auto sy = GetI18NCategory("System");
+			if (callback)
+				callback(Status::FAILURE, sy->T("Error: load undo state is from a different game"), cbUserData);
+			return false;
+		}
+
+		Path fn = GetSysDirectory(DIRECTORY_SAVESTATE) / LOAD_UNDO_NAME;
+		if (!fn.empty()) {
+			Load(fn, -2, callback, cbUserData); // Slot number is visual only, -2 will display special message (kinda have to find a better way)
```

Comment:
I think -2 is okay if we make it a `constexpr` at the top of the file somewhere after includes (I would not expose it in the .h.)

-[Unknown]